modules/instance_template/main.tf

```diff
@@ -99,6 +99,13 @@ resource "google_compute_instance_template" "tpl" {
     network            = var.network
     subnetwork         = var.subnetwork
     subnetwork_project = var.subnetwork_project
+    dynamic "access_config" {
+      for_each = var.public_ip ? [var.access_config] : []
+      content {
+        nat_ip       = lookup(access_config.value, "nat_ip", null)
+        network_tier = lookup(access_config.value, "network_tier", null)
```

Comment:
```suggestion
        network_tier           = access_config.value.network_tier
```

Reply:
I intentionally made `lookup(access_config.value, "network_tier", null)` and not `access_config.value.network_tier` because the `network_tier` can be absent.

Reply:
now, since the `network_tier` property is required, we can safely assign it directly `access_config.value.network_tier` without using lookup